nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3BinaryManager.java

```diff
@@ -445,9 +450,16 @@ public void storeFile(String digest, File file) throws IOException {
                 PutObjectRequest request;
                 if (!isEncrypted) {
                     request = new PutObjectRequest(bucketName, key, file);
-                    if (userServerSideEncryption) {
+                    if (useServerSideEncryption) {
                         ObjectMetadata objectMetadata = new ObjectMetadata();
-                        objectMetadata.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION);
+                        if (serverSideKMSKeyID != null) {
```

Comment:
Replace `serverSideKMSKeyID != null` with `isNotBlank(serverSideKMSKeyID)` as this comes from user input and we prefer to be lenient with spaces in that case.